### ISSUE TITLE
Learning rate schedules

### ISSUE THREAD

**Issue by CatStark** [jina-ai/finetuner#241]

**Comment by tadejsv** [comment]
These will not be implementable as callbacks, as their construction differs too much between frameworks, and in pytorch and paddle, they need to be either provided in optimizer initialization, or the optimizer provided in their initialization.

Given that, the direction I am taking is similar to what pytorch lightning is doing - users can provide a optimizer/lr scheduler factory function (also needed because pytorch optimizer needs model at initialization - if we move the model to a different device afterwards we break the optimizer).
